### PR TITLE
Updates imports and adds to api section of readme

### PR DESCRIPTION
**Pull request by dm36**
This pull request:

- Updates go imports to public github
- Adds to the API section
- Alphabetizes calls to checkParameterLength in checkOptions

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -135,6 +128,25 @@ The [LogDNA Ingestion Key](https://app.logdna.com/manage/profile) associated wit
...
+* Max Length: `32`
+
+The default environment passed along with every log sent through this instance.
+
+##### FlushInterval
+
+* _**Optional**_
+* Type: `time.duration`
+* Default: `5`
+* Values: `10`
```
Comment (smusali) [inline README.md:146]: How different are `Values` and `Default`?

File: README.md [hunk 2]
```diff
@@ -183,19 +206,64 @@ The default level passed along with every log sent through this instance.
...
+* Type: `int`
+* Default: `5`
+* Values: `10`
+
+MaxBufferLen sets the number of logs that are buffered before data is flushed and shipped to LogDNA.
+
+##### Meta
+
+* _**Optional**_
+* Type: `Meta`
```
Comment (smusali) [inline README.md:221]: I think this should be `struct`, no?

File: README.md (hunk 1 above)
Comment (dm36) [reply README.md:146]: Am following convention here: https://github.com/logdna/python.

Default is what the value defaults to if not provided, and values are possible values that the user could use for this field

Comment (smusali) [reply README.md:146]: oh, I see, @dm36, I'd say there is no need for `Values` - just leave `Default` if necessary

File: README.md (hunk 2 above)
Comment (dm36) [reply README.md:221]: it's a struct called Meta- and when declaring the variable you specify the type as Meta

Comment (smusali) [reply README.md:221]: We refer to the variable type when we say `Type` in this way and `meta` has a `Struct` type.

File: README.md (hunk 1 above)
Comment (dm36) [reply README.md:146]: I think default gives a view of what values are set to if not provided and values provides potential values that the user can set

File: README.md [hunk 3]
```diff
@@ -183,19 +206,64 @@ The default level passed along with every log sent through this instance.
...
+* Default: `5`
+* Values: `logging,golang`
+
+
+List of tags used to dynamically group hosts.
+
+#### Key
+
+* _**Required**_
+* Type: `string`
```
Comment (smusali) [inline README.md:238]: keep consistency: in one place it's `String` and in another place it's `string`

**Pull request merged**